### ISSUE TITLE
Updating pip vs. --disable-pip-version-check

### ISSUE THREAD

**Issue by Paebbels** [pyTooling/Actions#50]
`pip` is giving errors when it's outdated.

**There are two solutions:**
1. Update pip on every CI run with:  
   `pip install -U pip`
2. Run pip with an option to skip the version check:  
   `pip install --disable-pip-version-check -r requirements.txt`

Currently, we use solution 1.

**I see these advantages in solution 2:**
* less network traffic per job
* less setup (installation) time per job

**Besides an CLI option, this setting can be applied in various ways:**
* Configure pip from the command line
  ```bash
  pip config set global.disable-pip-version-check true
  ```
* Set an environment variable
  ```bash
  export PIP_DISABLE_PIP_VERSION_CHECK=1
  ```
* Use a config file `~/.config/pip/pip.conf`
  ```ini
  [global]
  disable-pip-version-check = True
  ```

Source: https://stackoverflow.com/a/46288945/3719459

@umarcor any disadvantages you see for solution 2?

**Comment by umarcor** [comment]
I think it's reasonable to use option 2 instead of 1. However, I don't have any preference with regard to using a configure option, the envvar or the configure file. I guess the envvar would be the easiest for users to override, if it wasn't because of the poor `env` support in GitHub Actions. Therefore, since we will need to provide an specific option/input for them to change the default, any solution will work.

**Comment by Paebbels** [comment]
This is implemented in #51.
